src/Files.App/Actions/FileSystem/BaseDeleteAction.cs

```diff
@@ -35,8 +35,8 @@ protected async Task DeleteItems(bool permanently)
 							: FilesystemItemType.Directory));
 
 			await context.ShellPage!.FilesystemHelpers.DeleteItemsAsync(items, settings.DeleteConfirmationPolicy, permanently, true);
-
 			await context.ShellPage.FilesystemViewModel.ApplyFilesAndFoldersChangesAsync();
+			Ioc.Default.GetRequiredService<SidebarViewModel>().SidebarPinnedModel.UpdateRecycleBinIcon();
```

Comment:
I'm setting this PR as a draft for now because of those lines appearing inside the actions. This is dirty and this needs to be changed, the UI update shouldn't be done inside the command logic.
 
The problem is that no event is fired when executing a command as far as I can tell, so I cannot hook the update on anything. 

I was thinking about creating a system that broadcasts events when executing a command - for example, the command name for starters. 

It's not a huge implementation, but maybe somebody would have an idea to bypass this need? Or would this command execution broadcast event system be interesting to have?

Reply:
Do you think we can move forward without updating the sidebar at runtime? True, the sidebar won't refresh until the next session but at least the desktop icon will update immediately.

Reply:
I'm working on another implementation that should solve both issues.

Reply:
Just adding this for reference #10367